### PR TITLE
Add axis validation in Glu.compute_output_spec for symbolic inputs

### LINKED ISSUE #23169: Glu does not validate axis parameter for symbolic inputs
## Description

`keras.ops.glu` does not validate the `axis` parameter when called with symbolic tensors (`keras.Input`), while it correctly raises errors for eager tensors.

## Expected
Both eager and symbolic inputs should validate `axis` and raise a `ValueError` when the axis is out of bounds.

This is similar to issues fixed in #22472 / #22595 for other ops (`flip`, `roll`, `trace`, etc.)

### PR DESCRIPTION
**Pull request by patminn107**
## Description

Adds missing axis validation in `Glu.compute_output_spec` to ensure invalid axis values are caught early for symbolic (KerasTensor) inputs, not just during eager execution.

## Changes

- **keras/src/ops/nn.py**: Added `canonicalize_axis(self.axis, len(output_shape))` in `Glu.compute_output_spec` (line was already imported)
- **keras/src/ops/nn_test.py**: Added test case that `knn.glu(x, axis=5)` raises `ValueError` on a rank-3 symbolic tensor

## Related

Closes #23169

## Contributor Agreement

**Please review our [AI-Assisted Contribution Policy](CONTRIBUTING.md#ai-assisted-contribution-policy) and check all boxes below before submitting your PR for review:**

- [x] I am a human, and not a bot.
- [x] I will be responsible for responding to review comments in a timely manner.
- [x] I will work with the maintainers to push this PR forward until submission.

**Note:** Failing to adhere to this agreement may result in your future PRs no longer being reviewed.

### CODE REVIEW

**Comment by patminn107** [comment]
Closing this in favor of #23171 (same changes, fresh branch)
